### DISCUSSION TITLE
Make API Access Requests Easier

### DISCUSSION THREAD

**Discussion by mycarrysun** [etsy/open-api#924] [Ideas]
I'd like to formally make a request that you change your API approval process. If you only need personal access you shouldn't need to be approved. This blocks up business channels and people cannot make their sales. API access is required for many integrations like Prestashop and can be really bad for business if you're standing around twiddling your thumbs cause you can't get API access.

Now, I understand if you need commercial access this is entirely different, I'm purely talking about personal access only.

I have been talking back and forth with support for 3 days trying to get an api key and they keep rejecting my submissions because I haven't "provided enough detail" even though I've filled the thing out with all the details of what I want to do.

I'm at my whit's end. Developer experience should be one of the top priorities for a company the size of Etsy.

**Comment by turboturtleD** [comment]
When you say you need this for personal access only, are you then intending to stick your API keys into an app that someone else developed for the Prestashop platform?

I think Etsy’s idea going forward is that *that app developer* needs to get a commercially approved app, and then you would just use the front end Auth flow that they build.  Then you would not need to procure your own key, or have an API key at all (but with all the same functionality).

If you are not the one doing the actual app development, you might have an uphill battle getting app approval.  Is the integration even v3 ready?  If so, the developer needs to get their app commercially approved - which would solve all your problems.

**Reply by mycarrysun** [reply]
No that's not what I am saying for my own personal use case. This API key will only live on my local machine as I'm just building a personal python script.

**Reply by mycarrysun** [reply]
Also I _am_ the one doing the actual app development.

**Comment by turboturtleD** [comment]
Ah!  Got it.  There are some other threads that talk about what is needed to "satisfy" the application reviewers.  I agree with you that from a dev perspective, it shouldn't be so hard to get a personal app approved.

Best luck - please share any tips for others trying to get approval when successful.
